src/Orleans/AssemblyLoader/AssemblyLoader.cs

```diff
@@ -120,7 +120,7 @@ internal class AssemblyLoader
             }
             catch (Exception exc)
             {
-                logger.Error(ErrorCode.Loader_LoadAndCreateInstance_Failure, exc.Message, exc);
+                logger.Error((int)ErrorCode.Loader_LoadAndCreateInstance_Failure, exc.Message, exc);
```

Comment:
It's probably worth having internal extension methods that take the ErrorCode enum, so all of our code still internally use the original approach, which is valid and (a little) less prone to collisions.